### PR TITLE
DxDiagram upd documentation

### PR DESCRIPTION
**Pull request by LanaDX**

### CODE REVIEW

File: api-reference/10 UI Components/dxDiagram/1 Configuration/customShapes/textHeight.md [hunk 1]
```diff
@@ -11,3 +11,4 @@ This property specifies a fraction from 0 to 1 that represents the height of a s
```
Comment (albertov05) [inline api-reference/10 UI Components/dxDiagram/1 Configuration/customShapes/textHeight.md:11]: `This property specifies a fraction from 0 to 1 that represents the height of a shape's text in relation to the shape height.`

Please rewrite without using "represents".

File: api-reference/10 UI Components/dxDiagram/1 Configuration/customShapes/textHeight.md [hunk 2]
```diff
@@ -11,3 +11,4 @@ This property specifies a fraction from 0 to 1 that represents the height of a s
 
 The absolute text height is calculated by multiplying the **textHeight** value by the shape height.
 
+If the **textHeight** property is set to `0` (zero) or `undefined` and the [autoSizeEnabled](/Documentation/ApiReference/UI_Components/dxDiagram/Configuration/nodes/#autoSizeEnabled) property is set to `true`, the **textHeight** property value is considered equal to `1`.
```
Comment (albertov05) [inline api-reference/10 UI Components/dxDiagram/1 Configuration/customShapes/textHeight.md]:
```suggestion
If the **textHeight** property is set to `0` (zero) or `undefined` and the [autoSizeEnabled](/Documentation/ApiReference/UI_Components/dxDiagram/Configuration/nodes/#autoSizeEnabled) property is set to `true`, the **textHeight** property value is equal to `1`.
```

File: api-reference/10 UI Components/dxDiagram/1 Configuration/customShapes/textWidth.md [hunk 3]
```diff
@@ -9,4 +9,6 @@ Specifies the shape text container's width.
 ---
 This property specifies a fraction from 0 to 1 that represents the shape text's width in relation to the shape width. If the **textWidth** property is set to 1, the text width is equal to the shape width.
```
Comment (albertov05) [inline api-reference/10 UI Components/dxDiagram/1 Configuration/customShapes/textWidth.md]: `This property specifies a fraction from 0 to 1 that represents the shape text's width in relation to the shape width.`

Please rewrite without using "represents".

File: api-reference/10 UI Components/dxDiagram/1 Configuration/customShapes/textWidth.md [hunk 4]
```diff
@@ -9,4 +9,6 @@ Specifies the shape text container's width.
...
 
-The absolute text width is calculated by multiplying the **textWidth** value by the shape width.
\ No newline at end of file
+The absolute text width is calculated by multiplying the **textWidth** value by the shape width.
+
+If the **textWidth** property is set to `0` (zero) or `undefined` and the [autoSizeEnabled](/Documentation/ApiReference/UI_Components/dxDiagram/Configuration/nodes/#autoSizeEnabled) property is set to `true`, the **textWidth** property value is considered equal to `1`.
```
Comment (albertov05) [inline api-reference/10 UI Components/dxDiagram/1 Configuration/customShapes/textWidth.md]:
```suggestion
If the **textWidth** property is set to `0` (zero) or `undefined` and the [autoSizeEnabled](/Documentation/ApiReference/UI_Components/dxDiagram/Configuration/nodes/#autoSizeEnabled) property is set to `true`, the **textWidth** property value is equal to `1`.
```

File: api-reference/10 UI Components/dxDiagram/1 Configuration/hasChanges.md [hunk 5]
```diff
@@ -8,7 +8,7 @@ default: false
 Indicates whether diagram content has been changed.
 
 ---
-The **hasChanges** property is set to **true** on any diagram change. You can use this property together with the [optionChanged](/api-reference/10%20UI%20Components/DOMComponent/4%20Events/optionChanged.md '/Documentation/ApiReference/UI_Components/dxDiagram/Events/#optionChanged') event to implement diagram content autosave.
+The **hasChanges** property is set to **true** on any diagram change. You can use this property together with the [optionChanged](/api-reference/10%20UI%20Components/DOMComponent/4%20Events/optionChanged.md '/Documentation/ApiReference/UI_Components/dxDiagram/Events/#optionChanged') event to implement diagram content autosave. After saving, set the **hasChanges** property to `false`.
```
Comment (albertov05) [inline api-reference/10 UI Components/dxDiagram/1 Configuration/hasChanges.md]:
```suggestion
The **hasChanges** property is set to **true** on any diagram change. You can use this property together with the [optionChanged](/api-reference/10%20UI%20Components/DOMComponent/4%20Events/optionChanged.md '/Documentation/ApiReference/UI_Components/dxDiagram/Events/#optionChanged') event to implement diagram content autosave. After the content is saved, set the **hasChanges** property to `false`.
```

File: concepts/05 UI Components/Diagram/10 Data Binding/40 Optional Binding Expressions.md [hunk 6]
```diff
@@ -162,6 +162,8 @@ The Diagram allows you to bind a number of shape and connector visual properties
```
Comment (albertov05) [inline concepts/05 UI Components/Diagram/10 Data Binding/40 Optional Binding Expressions.md:162]: Please change line 158 (above) as follows:

`<td>An edge's z-index.</td>`

File: concepts/05 UI Components/Diagram/10 Data Binding/40 Optional Binding Expressions.md [hunk 7]
```diff
@@ -162,6 +162,8 @@ The Diagram allows you to bind a number of shape and connector visual properties
 
 You can set a binding property to a name of a data source field that provides item values, or to an expression that returns a constant value or calculates a value in runtime based on conditions.
```
Comment (albertov05) [inline concepts/05 UI Components/Diagram/10 Data Binding/40 Optional Binding Expressions.md]: We should rewrite this sentence without using "provides". Can we say "supplies"?

Also, does only one data source field supply item values?<br>If so, then "a name of a data source field" should be "the name of the data source field".

File: api-reference/10 UI Components/dxDiagram/1 Configuration/customShapes/textWidth.md (hunk 3 above)
Comment (LanaDX) [reply api-reference/10 UI Components/dxDiagram/1 Configuration/customShapes/textWidth.md]: fixed

File: concepts/05 UI Components/Diagram/10 Data Binding/40 Optional Binding Expressions.md (hunk 6 above)
Comment (LanaDX) [reply concepts/05 UI Components/Diagram/10 Data Binding/40 Optional Binding Expressions.md:162]: updated

File: api-reference/10 UI Components/dxDiagram/1 Configuration/customShapes/textHeight.md (hunk 1 above)
Comment (LanaDX) [reply api-reference/10 UI Components/dxDiagram/1 Configuration/customShapes/textHeight.md:11]: fixed

File: concepts/05 UI Components/Diagram/10 Data Binding/40 Optional Binding Expressions.md (hunk 7 above)
Comment (LanaDX) [reply concepts/05 UI Components/Diagram/10 Data Binding/40 Optional Binding Expressions.md]: updated

**Review by AnnaKondratova — APPROVED** [review]

File: api-reference/10 UI Components/dxDiagram/1 Configuration/customShapes/textWidth.md (hunk 3 above)
Comment (albertov05) [reply api-reference/10 UI Components/dxDiagram/1 Configuration/customShapes/textWidth.md]: Thanks

File: concepts/05 UI Components/Diagram/10 Data Binding/40 Optional Binding Expressions.md (hunk 6 above)
Comment (albertov05) [reply concepts/05 UI Components/Diagram/10 Data Binding/40 Optional Binding Expressions.md:162]: Thanks

File: concepts/05 UI Components/Diagram/10 Data Binding/40 Optional Binding Expressions.md (hunk 7 above)
Comment (albertov05) [reply concepts/05 UI Components/Diagram/10 Data Binding/40 Optional Binding Expressions.md]: Thanks

**Pull request merged**